app/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSender.java

```diff
@@ -16,7 +28,66 @@ private ReportStreamLabOrderSender() {}
 
     @Override
     public void sendOrder(final LabOrder<?> order) {
-        // String json = FHIR.stringify(order.getUnderlyingOrder())
-        // reportStream.sendRequestBody(json)
+        String json = fhir.encodeResourceToJson(order.getUnderlyingOrder());
+        String bearerToken = requestToken();
+        sendRequestBody(json, bearerToken);
+    }
+
+    public void sendRequestBody(@NotNull String json, @NotNull String bearerToken) { // url param?
+        String res;
+        Map<String, String> headers =
+                Map.of(
+                        "Authorization", "Bearer" + bearerToken,
+                        "client", "flexion",
+                        "Content-Type", "application/hl7-v2");
+        try {
+            res = client.post(this.STAGING, headers, json); // what to do with response?
+        } catch (IOException e) {
+            // TODO exception handling
+        }
+    }
+
+    protected String requestToken() {
+        String senderToken = null;
+        String token = "";
+        String body;
+        String sender = "flexion.etor-service-sender";
+        String keyId = "flexion.etor-service-sender";
+        Map<String, String> headers = Map.of("Content-Type", "application/x-www-form-urlencoded");
+        try {
+            senderToken = jwt.generateSenderToken(sender, this.STAGING_AUTH, "pemKey", keyId, 300);
+            body = composeRequestBody(senderToken);
+            String rsResponse = client.get(this.STAGING_AUTH, headers, body);
+            // TODO response handling when it fails?
+            token = extractToken(rsResponse);
+        } catch (Exception e) {
+            // TODO exception handling
+        }
+        return token;
+    }
+
+    protected String extractToken(String responseBody) {
+        String key = "access_token";
+        Map<String, String> value = null;
+        try {
+            value = jackson.convertToObject(responseBody, Map.class);
+        } catch (FormatterProcessingException e) {
+            // TODO exception handling
+        }
+        return value.get(key);
```

Comment:
Eh, probably no need to have a separate variable for the key.

```suggestion
        Map<String, String> value = null;
        try {
            value = jackson.convertToObject(responseBody, Map.class);
        } catch (FormatterProcessingException e) {
            // TODO exception handling
        }
        return value.get("access_token");
```